src/coreclr/System.Private.CoreLib/src/System/GC.cs

```diff
@@ -428,6 +428,8 @@ public static GCNotificationStatus WaitForFullGCApproach(int millisecondsTimeout
             return (GCNotificationStatus)_WaitForFullGCApproach(millisecondsTimeout);
         }
 
+        public static GCNotificationStatus WaitForFullGCApproach(TimeSpan timeout) => WaitForFullGCApproach(checked((int)timeout.TotalMilliseconds));
```

Comment:
Using checked will result in an overflow exception.  Elsewhere we accept TimeSpan and translate that into an int, for out of range values we use ArgumentOutOfRange.

This goes for all such use in this PR.

Reply:
@stephentoub Thread.Sleep uses a WaitHandle overload <https://github.com/dotnet/runtime/blob/e4fb7303f7be173f5434d711d42b4bba706d0ad0/src/libraries/System.Private.CoreLib/src/System/Threading/Thread.cs#L507>

I think we should use this. What do you think?

Reply:
I have done this once inside the CoreLib now. And per assembly this implemented again (in the long run, we should put this in a central location to reduce code duplication).